route/build/cpl/RtmMod.F90

```diff
@@ -292,59 +295,74 @@ SUBROUTINE route_run(rstwr)
 
     delt_coupling = coupling_period*60.0*60.0*24.0
     if (first_call) then
-       nsub_save = 1
-       delt_save = dt
-       if (masterproc) write(iulog,'(2a,g20.12,a)') trim(subname),' mizuRoute coupling period ',delt_coupling, '[sec]'
+      nsub_save = 1
+      delt_save = dt
+      if (masterproc) write(iulog,'(2a,g20.12,a)') trim(subname),' mizuRoute coupling period ',delt_coupling, '[sec]'
     end if
 
     !-------------------------------------------------------
     ! Initialize mizuRoute history handler and fields
     !-------------------------------------------------------
     call t_startf('mizuRoute_histinit')
+
     call main_new_file(ierr, cmessage)
-    call t_stopf('mizuRoute_histinit')
+    if(ierr/=0)then; call shr_sys_abort(trim(subname)//trim(cmessage)); endif
 
-    rtmCTL%discharge = 0._r8
-    rtmCTL%volr      = 0._r8
-    rtmCTL%flood     = 0._r8
+    call t_stopf('mizuRoute_histinit')
 
-    !-----------------------------------
-    ! Compute irrigation flux based on demand from clm
+    !----------------------------------------------------------
+    ! Mapping CLM imported fluxes to mizuRoute HRUs or Reaches
+    !-----------------------------------------------------------
+    ! --- Mapping irrigation demand [mm/s] at HRU to river reach
     ! Must be calculated before volr is updated to be consistent with lnd
-    ! Just consider land points and only remove liquid water
-    !-----------------------------------
+    call t_startf('mizuRoute_mapping_irrig')
+
+    allocate(array_temp(rtmCTL%lnumr))
```

Comment:
As we talked about last week. Most of the allocates here, should be move to route_ini. Then you won't have to check if they are allocated. It's a good practice to have a clean method to deallocate the data as well, but currently there isn't such a method for this module (something like a route_fini subroutine). It's OK to not have a deallocate if the memory is held for the entire length of the program.

Reply:
Now memory allocated in preparation step and so no need to check if they are allocated.  commit [e7a5570](https://github.com/ESCOMP/mizuRoute/pull/288/commits/e7a557076c24caba84f003e4dc3ee4d2fa615376)